### PR TITLE
Fix py2

### PR DESCRIPTION
**Pull request by manodeep**
Fixing #214

### CODE REVIEW

**Comment by manodeep** [comment]
Explanation from @1313e as to why installing in editable mode is necessary on TRAVIS:

> The reason for it is that on Travis CI (specifically on Linux), the working directory is set to the root of the package. If you install it the normal way and then try to import it, the local package takes priority over the installed package (see my notes on how Python imports work from last year). As you haven't built that, it cannot find any files and crashes.

File: CHANGES.rst [hunk 1]
```diff
@@ -21,11 +20,27 @@ Enhancements
...
+
+
+
 Bug fixes
 ----------
 - Installation does not require python(3)-config anymore [#209, #211]
 - Better handling of terminal colours for unknown terminals [#209]
-
+- Prevent incorrect calculations with periodic boundaries for large ratios of
+(zmax, Rmax) to Lbox [#210]
```
Comment (lgarrison) [inline CHANGES.rst]: Is this supposed to be in this PR?

File: .travis.yml [hunk 2]
```diff
@@ -103,12 +103,15 @@ install:
...
+  - conda install -q --yes numpy=$NUMPY_VERSION || conda install -q --yes -c conda-forge numpy=$NUMPY_VERSION
   - conda install -q --yes -c conda-forge gsl
+  - python -m pip install --upgrade pip
+  - python -m pip install --ignore-installed certifi --upgrade
+  - python -m pip install sphinx>=1.8
   - if [[ "${INSTALL_LIBGFORTRAN}" == "TRUE" ]]; then conda install -q --yes libgfortran ; fi
   - make -r CC=$COMPILER
   - make install CC=$COMPILER
-  - python setup.py install
+  - python -m pip install -e .
```
Comment (lgarrison) [inline .travis.yml]: `pip install -e` is a great trick, I can't believe I'm just learning about it now!

File: mocks/python_bindings/Makefile [hunk 3]
```diff
@@ -63,16 +63,16 @@ $(VPF_mocks_DIR)/lib$(VPF_mocks_LIB).a: $(VPF_mocks_DIR)/*.c $(VPF_mocks_DIR)/*.
...
+# ifeq ($(FIX_PYTHON_LINK), 1)
+# 	@{ \
+# 		CURRENT_PYTHON_LIB=`otool -L $(PYTHON_EXTN) | grep -i python | cut -d " " -f1 | grep -v $(PROJECT) | xargs` ; \
+# 		PYTHON_LIB_NAME=$(PYTHON_LIB_BASE).dylib ; \
+# 		LINK_PYTHON_LIB=$(PYTHON_LIBDIR)/$$PYTHON_LIB_NAME ;\
+# 		if [[ "$$CURRENT_PYTHON_LIB" != "$$LINK_PYTHON_LIB" ]] ; then \
+# 		install_name_tool -change $$CURRENT_PYTHON_LIB $$LINK_PYTHON_LIB $(PYTHON_EXTN); \
+# 		fi ;\
+# 	}
+# endif
```
Comment (lgarrison) [inline mocks/python_bindings/Makefile]: Ready to be removed, or do you still want to keep this around?

File: theory/python_bindings/Makefile [hunk 4]
```diff
@@ -70,18 +70,18 @@ $(VPF_DIR)/lib$(VPF_LIB).a: $(VPF_DIR)/*.c $(VPF_DIR)/*.c.src $(VPF_DIR)/*.h.src
...
+# ifeq ($(FIX_PYTHON_LINK), 1)
+# 	@{ \
+# 		CURRENT_PYTHON_LIB=`otool -L $(PYTHON_EXTN) | grep -i python | cut -d " " -f1 | grep -v $(PROJECT) | xargs` ; \
+# 		PYTHON_LIB_NAME=$(PYTHON_LIB_BASE).dylib ; \
+# 		LINK_PYTHON_LIB=$(PYTHON_LIBDIR)/$$PYTHON_LIB_NAME ;\
+# 		if [[ "$$CURRENT_PYTHON_LIB" != "$$LINK_PYTHON_LIB" ]] ; then \
+# 		install_name_tool -change $$CURRENT_PYTHON_LIB $$LINK_PYTHON_LIB $(PYTHON_EXTN); \
+# 		fi ;\
+# 	}
+# endif
```
Comment (lgarrison) [inline theory/python_bindings/Makefile]: Same question here

File: CHANGES.rst (hunk 1 above)
Comment (manodeep) [reply CHANGES.rst]: This was in the release notes but not in the CHANGES.rst  - so I added it in.

File: .travis.yml (hunk 2 above)
Comment (manodeep) [reply .travis.yml]: Yup - it's very handy. It's the recommended way for testing while developing/tinkering

File: mocks/python_bindings/Makefile (hunk 3 above)
Comment (manodeep) [reply mocks/python_bindings/Makefile]: Might as well remove them

**Pull request merged**